src/book/book.rs

```diff
@@ -75,10 +75,10 @@ fn create_missing(src_dir: &Path, summary: &Summary) -> Result<()> {
 /// [`iter()`]: #method.iter
 /// [`for_each_mut()`]: #method.for_each_mut
 #[derive(Debug, Clone, Default, PartialEq, Serialize, Deserialize)]
+#[non_exhaustive]
```

Comment:
this is a breaking change, it was already made in https://github.com/rust-lang/mdBook/pull/1854 but it is waiting to be added into a new major release